### PR TITLE
feat(conductor): remove disable empty block execution config setting

### LINKED ISSUE #555: Remove disable empty block execution config setting
To make sure that the Conductor sync works predictably and that some of the recent optimizations work properly, it has been determined that removing the option to not execute empty blocks should be removed in Conductor.

### PR DESCRIPTION
**Pull request by sambukowski**
## Summary
The 'DISABLE_EMPTY_BLOCK_EXECUTION` config option from Conductor.

## Background
To make sure that the Conductor sync works reliably, it was determined that the option to disable empty block execution should be removed. The makes all block times regular and some of the optimizations to the Conductor sync more efficient.

## Changes
- `DISABLE_EMPTY_BLOCK_EXECUTION` removed
- Updated `astria_conductor::executor::Executor::execute_block` to no longer return an option.
- Updated tests to make sure that executing empty blocks and blocks with transactions work the same way

## Testing
passes cargo code test
manual testing using dev-cluster

## Breaking Changelist
- Requires updates to charts in dev cluster

closes #555

### CODE REVIEW

File: crates/astria-conductor/src/executor/mod.rs [hunk 1]
```diff
@@ -214,14 +209,6 @@ impl Executor {
     /// execution block hash.
     /// if there are no relevant transactions in the SequencerBlock, it returns None.
     async fn execute_block(&mut self, block: SequencerBlockSubset) -> Result<Option<Block>> {
```
Comment (noot) [inline crates/astria-conductor/src/executor/mod.rs:211]: change this to just return `Result<Block>` and update the comment above

**Review by noot — APPROVED** [review]
please address comment before merging!!

File: crates/astria-conductor/src/executor/mod.rs (hunk 1 above)
Comment (SuperFluffy) [reply crates/astria-conductor/src/executor/mod.rs:211]: ack, this signature can now be simplified

**Review by joroshiba — APPROVED** [review]

File: crates/astria-conductor/src/executor/mod.rs (hunk 1 above)
Comment (sambukowski) [reply crates/astria-conductor/src/executor/mod.rs]: updated!

**Pull request merged**